http/client.go

```diff
@@ -15,9 +16,23 @@ import (
 	spy "github.com/j0hnsmith/connspy/net"
 )
 
-// NewClient returns a http.Client that will output all http data to stdout. The client has various default timeouts,
-// call with nil values to use them, otherwise pass arguments to customise.
+// NewClient returns a http.Client that will output all http data to stderr.
+// The client has various default timeouts, call with nil values to use them, otherwise pass arguments to customise.
 func NewClient(dialer *net.Dialer, transport *http.Transport) *http.Client {
+	client := clientToWriter(dialer, transport, os.Stderr)
```

Comment:
Rather than adding to the public api, how about this, seems like it would achieve the same goal?

```
func NewClient(dialer *net.Dialer, transport *http.Transport, writer io.Writer) *http.Client {
    if writer == nil {
        writer = os.Stderr
    }
...

Reply:
I didn't want to change the original API, so I added it as a new method. This would break the API right, as it'd fail to compile if people hadn't provided the writer parameter?